### PR TITLE
Update exceptions.json adding RetroDECK xdg-access

### PR DESCRIPTION
**Pull request by XargonWan**
From https://github.com/flathub/flathub/issues/4786

### CODE REVIEW

File: flatpak_builder_lint/staticfiles/exceptions.json [hunk 1]
```diff
@@ -1147,6 +1147,10 @@
     "net.lutris.Lutris": {
         "finish-args-flatpak-spawn-access": "the app predates this linter rule"
     },
+    "net.retrodeck.retrodeck": {
+        "appid-uses-code-hosting-domain": "app-id predates this linter rule",
```
Comment (bbhtt) [inline flatpak_builder_lint/staticfiles/exceptions.json:1151]: net.retrodeck doesn't seem a code hosting domain?

File: flatpak_builder_lint/staticfiles/exceptions.json [hunk 2]
```diff
@@ -1147,6 +1147,10 @@
...
+        "finish-args-unnecessary-xdg-data-access": "https://github.com/flathub/flathub/issues/4786",
```
Comment (bbhtt) [inline flatpak_builder_lint/staticfiles/exceptions.json:1152]: The current usage is for gtk-3.0 folder, why is that needed? Also, can you please put the actual reason here, not the link.

File: flatpak_builder_lint/staticfiles/exceptions.json (hunk 1 above)
Comment (hfiguiere) [reply flatpak_builder_lint/staticfiles/exceptions.json:1151]: also don't lie. this doesn't predate a linter rule, because there is none against that app-id.

**Comment by hfiguiere** [comment]
Closing as per matrix conversation and https://github.com/flathub/flatpak-builder-lint/pull/229#discussion_r1426685382

**Comment by XargonWan** [comment]
Turns out that that arg is needed for displaying the dark theme on GTK.
It seems to be a Valve side bug that is under review: https://github.com/ValveSoftware/SteamOS/issues/803
SteamOS 3 doesn't have GTK portal.

File: flatpak_builder_lint/staticfiles/exceptions.json (hunk 1 above)
Comment (hfiguiere) [reply flatpak_builder_lint/staticfiles/exceptions.json:1151]: this need to be resolved, it's not an option

File: flatpak_builder_lint/staticfiles/exceptions.json [hunk 3]
```diff
@@ -1147,6 +1147,10 @@
     "net.lutris.Lutris": {
         "finish-args-flatpak-spawn-access": "the app predates this linter rule"
     },
+    "net.retrodeck.retrodeck": {
+        "appid-uses-code-hosting-domain": "app-id predates this linter rule",
+        "finish-args-unnecessary-xdg-data-access": "SteamOS is the primary target of the application and it does not ship xdg-desktop-portal-gtk: https://github.com/flathub/flathub/issues/4786"
```
Comment (TingPing) [inline flatpak_builder_lint/staticfiles/exceptions.json:1152]:
```suggestion
        "finish-args-unnecessary-xdg-data-access": "SteamOS is the primary target of the application and it does not ship an xsettings daemon or xdg-desktop-portal-gtk: https://github.com/ValveSoftware/SteamOS/issues/803"
```

**Review by bbhtt — APPROVED** [review]
LGTM, because the app primarily targets steam deck which lacks necessary desktop integration to load GTK themes.
